roles/edpm_pre_adoption_validation/tasks/main.yml

```diff
@@ -0,0 +1,56 @@
+---
+# Copyright 2023 Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Validate that hostname for the EDPM openstack services do not change
+  become: true
+  tags:
+    - adoption
+    - validation
+  block:
+    - name: Check if old nova.conf exists
+      ansible.builtin.stat:
+        path: "{{ edpm_pre_adoption_validation_old_nova_config }}"
+      register: nova_old_config_file
+
+    - name: Compare old nova service host to {{ canonical_hostname }}
+      when: nova_old_config_file.stat.exists
+      ansible.builtin.shell: |
+        set -o pipefail
+        egrep -e '^host.*=' {{ edpm_pre_adoption_validation_old_nova_config }} | cut -d '=' -f2 | sed 's/ //g'
+      register: nova_old_config_output
+      changed_when: true
+      failed_when: nova_old_config_output.stdout != canonical_hostname
```

Comment:
nit this shoudl be changed_when: false
https://ansible.readthedocs.io/projects/lint/rules/no-changed-when/

changed_when: true implies we modifed something as a side effect fo the shell command.

failed_when: nova_old_config_output.stdout != canonical_hostname

is a nice way to do that test +1

Reply:
fixed the changed_when clause here and below